src/Stratis.Bitcoin/Consensus/ValidationResults/ConnectBlocksResult.cs

```diff
@@ -16,13 +16,40 @@ internal class ConnectBlocksResult : ValidationResult
 
         public ChainedHeader LastValidatedBlockHeader { get; set; }
 
-        public ConnectBlocksResult(bool succeeded, bool consensusTipChanged = true, List<int> peersToBan = null, string banReason = null, int banDurationSeconds = 0)
+        private ConnectBlocksResult() { }
+
+        public static ConnectBlocksResult Failed(bool consensusTipChanged = true)
```

Comment:
I think if we have `Success` then it should be `Failure` rather than `Failed` (my bad for suggesting it).